tensorboard/examples/plugins/example_basic/README.md

```diff
@@ -2,51 +2,159 @@
+    - `disable_reload`: Whether to disable the reload button and auto-reload timer. A `bool`; defaults to `False`.
+    - `es_module_path`: ES module to use as an entry point to this plugin. A `str` that is a key in the result of `get_plugin_apps()`.
+    - `remove_dom`: Whether to remove the plugin DOM when switching to a different plugin. A `bool`; defaults to `False`.
+    - `tab_name`: Name to show in the menu item for this dashboard within the navigation bar. May differ from the plugin name. An optional `str`, that defaults to the plugin name.
+
+If your plugin requires parsing flags or custom loading, consider defining a `TBLoader` as the entry point. Doing so is optional.
+
+For example:
+
+```python
+class MyLoader(base_plugin.TBLoader):
+  ### Plugin Setup
+
+  def define_flags(self, parser):
+    parser.add_argument_group('custom').add_argument('--enable_my_extras')
+
+  def fix_flags(self, flags):
+    if flags.enable_my_extras:
+      raise ValueError('Extras not ready')
+
+  def load(self, context):
+    return MyPlugin(context)
+```
+
+#### TBLoader
+  - `define_flags(parser)`: Optional method that takes an argparse.Namespace and exposes command-line flags. Please prefix flags with the name of the plugin to avoid collision.
+  - `fix_flags(flags)`: Optional method needed to fix or sanitize command-line flags.
+  - `load(context)`: Required method that takes a TBContext and returns a TBPlugin instance.
+
+It's recommended that plugins using flags call the `parser.add_argument_group(plugin_name)`. To learn more about the flag definition, see [docs](https://docs.python.org/library/argparse.html#adding-arguments)
+
+
+
+## Reading data from event files
+
+On instantiation, a plugin is provided a [`PluginEventMultiplexer`] object from which to read data. The `PluginRunToTagToContent` method on the multiplexer returns a dictionary containing all run–tag pairs and associated summary metadata for your plugin.
 
 Plugins are not technically restricted from arbitrary file system and network access, but we strongly recommend using the multiplexer exclusively. This abstracts over the filesystem (local or remote), provides a consistent user experience for runs and tags across plugins, and is optimized for TensorBoard read patterns.
 
 [`PluginEventMultiplexer`]: https://github.com/tensorflow/tensorboard/blob/master/tensorboard/backend/event_processing/plugin_event_multiplexer.py
 
+Example use of the multiplexer:
+```python
+class MyPlugin(base_plugin.TBPlugin):
+  def __init__(self, context):
+    self.multiplexer = context.multiplexer
+
+  def collect_data(self):
+    """
+    {runName: { images: [tag1, tag2, tag3],
+                scalarValues: [tagA, tagB, tagC],
+                histograms: [tagX, tagY, tagZ],
+                compressedHistograms: [tagX, tagY, tagZ],
+                graph: true, meta_graph: true}}
+    """
+    runs = self.multiplexer.Runs()
```

Comment:
I don't think it is kosher to call this method. You should expect FE to have the right set of runs and just take the name of the run as the HTTP API input. IDK how `collect_data` is supposed to be used but I think it should just exemplify `multiplexer.Tensors(run, tag)`.

Reply:
I think there are a couple cases:
1) serve a list of runs by adding a backend route that serves results from `multiplexer.Runs()`
2) use `multiplexer.Runs()` to preprocess / filter data before making it available for serving to frontend

I think that when we have a plugin library, case 1) should be replaced by calling `TB.getRuns()` on the FE, since it saves a request to the server.  But if an author wishes to, say, serve a list of tags instead of a list of runs, is it reasonable to make this list in their backend using the multiplexer?

For case 2), is it reasonable for authors to use it for preprocessing?  To transform data in some way that depends on the run name?  e.g. `if runName.includes('test/') /* ... */ else if runName.includes('train') /* ... */`

Reply:
Alternatively, we could remove this from docs, and update our current example (which uses `multiplexer.Runs()` with a TODO saying: once issue #X is resolved, use the plugin library.